### PR TITLE
a new class Property

### PR DESCRIPTION
**Pull request by jjren**
1. In class `TdMpsJob`,  `dump_type` is added, which could dump results in `JSON`(`dump_type=".json"`) or `numpy npz`(`dump_type=".npz"`, default) format to solve that JSON does not support numpy array or complex number.

2. A new class `Property` allows user-defined MPO to be calculated during TD simulation or for one single state (refer to `thermalprop` or test).

3. `property/ops.py` contains operators for property calculation. 
Only the electron phonon static correlation function operator is available now.

### CODE REVIEW

File: renormalizer/mps/tests/test_evolve.py [hunk 1]
```diff
@@ -71,7 +71,9 @@ def test_tdvp_vmf(init_state, with_mu, force_ovlp, atol):
 def test_tdvp_cmf(init_state):
     mps = init_state.copy()
     mps.evolve_config  = EvolveConfig(EvolveMethod.tdvp_mu_cmf)
-    check_result(mps, mpo, 0.01, 0.5, 5e-4)
+    for flag in [True, False]:
```
Comment (liwt31) [inline renormalizer/mps/tests/test_evolve.py]: It'll be more consistent if the option is moved to the decorator `pytest.mark.parametrize`

File: renormalizer/property/ops.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+import numpy as np
+
+
+def e_ph_static_correlation(mol_list: MolList, imol:int =0, jph:int =0,
+        periodic:bool =False, name:str="S"):
+    '''
+    construct the electron-phonon static correlation operator in polaron problem
+    if periodic:
+        S_(m, jph) = \frac{1}{D} \sum_n \langle x_{m+n,jph} a_n^\dagger a_n \rangle
```
Comment (liwt31) [inline renormalizer/property/ops.py]: What is the definition of `D`? Is this implying periodic *homogeneous* model?

File: renormalizer/property/property.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            mps_conjs:Union[Dict[str, Mps], Dict[str, MpDm], Mps, MpDm, None] =None):
+        '''
+        calculate all the properties with same mps or mps_conj
+        or 
+        calculate each property with different {prop_str:mps}, {prop_str:mps_conj}
+        '''
+        for prop_str in self.prop_strs:
+            
+            # pick ket
+            if mpss.is_mps or mpss.is_mpdm:
```
Comment (liwt31) [inline renormalizer/property/property.py]: If `Dict[str, Mps]` is passed in as `mpss` then this line would result in an `AttributeError`.  I suggest deleting the logic for `Dict` and create a new method when the necessity comes.

Comment (jjren) [reply renormalizer/property/property.py]: Yes. It's a bug.

File: renormalizer/property/ops.py (hunk 2 above)
Comment (jjren) [reply renormalizer/property/ops.py]: D is the displacement. I will add a reference paper.

File: renormalizer/mps/tests/test_evolve.py (hunk 1 above)
Comment (jjren) [reply renormalizer/mps/tests/test_evolve.py]: fixed.

File: renormalizer/property/property.py (hunk 3 above)
Comment (jjren) [reply renormalizer/property/property.py]: fixed

**Pull request merged**